kuma/users/tests/test_views.py

```diff
@@ -180,6 +180,136 @@ def test_get_ban_user_view(self):
         ok_(testuser.get_absolute_url() in resp['Location'])
 
 
+@pytest.mark.bans
+class BanAndCleanupTestCase(UserTestCase):
+    localizing_client = True
+
+    def test_ban_permission(self):
+        """The ban permission controls access to the ban and cleanup view."""
+        admin = self.user_model.objects.get(username='admin')
+        testuser = self.user_model.objects.get(username='testuser')
+
+        # testuser doesn't have ban permission, can't ban.
+        self.client.login(username='testuser',
+                          password='testpass')
+        ban_url = reverse('users.ban_user_and_cleanup',
+                          kwargs={'user_id': admin.id})
+        resp = self.client.get(ban_url)
+        eq_(302, resp.status_code)
+        ok_(str(settings.LOGIN_URL) in resp['Location'])
+        self.client.logout()
+
+        # admin has ban permission, can ban.
+        self.client.login(username='admin',
+                          password='testpass')
+        ban_url = reverse('users.ban_user_and_cleanup',
+                          kwargs={'user_id': testuser.id})
+        resp = self.client.get(ban_url)
+        eq_(200, resp.status_code)
+
+    def test_ban_nonexistent_user(self):
+        """GETs to ban_user_and_cleanup for nonexistent user return 404."""
+        testuser = self.user_model.objects.get(username='testuser')
+
+        # GET request
+        self.client.login(username='admin',
+                          password='testpass')
+        ban_url = reverse('users.ban_user_and_cleanup',
+                          kwargs={'user_id': testuser.id})
+        testuser.delete()
+        resp = self.client.get(ban_url)
+        eq_(404, resp.status_code)
+
+
+@pytest.mark.bans
+class BanUserAndCleanupSummaryTestCase(UserTestCase):
+    localizing_client = True
+
+    def test_ban_nonexistent_user(self):
+        """POSTs to ban_user_and_cleanup for nonexistent user return 404."""
+        testuser = self.user_model.objects.get(username='testuser')
+
+        # POST request
+        self.client.login(username='admin',
+                          password='testpass')
+        ban_url = reverse('users.ban_user_and_cleanup_summary',
+                          kwargs={'user_id': testuser.id})
+        testuser.delete()
```

Comment:
There's some common setup that could be refactored to a method or the ``setUp`` method.